### PR TITLE
fix(folders): [#FOR-635] forbid user to access not owned folder infos

### PR DESCRIPTION
**Pull request by qperie**
## Describe your changes
- Block the user to access folder not owned by API call
- Rename Model interface to IModel
- Add helpers for model use

## Issue ticket number and link
FOR-635 : https://jira.support-ent.fr/browse/FOR-635

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: formulaire/src/main/java/fr/openent/formulaire/service/impl/DefaultFolderService.java [hunk 1]
```diff
@@ -39,6 +43,16 @@ public void get(String folderId, Handler<Either<String, JsonObject>> handler) {
...
+    @Override
+    public Future<JsonObject> get(String folderId, UserInfos user) {
+        Promise<JsonObject> promise = Promise.promise();
+
+        String errorMessage = "[Formulaire@DefaultFolderService::get] Fail to get folder with id " + folderId + " : ";
+        get(folderId, FutureHelper.handlerEither(promise, errorMessage));
+
+        return promise.future();
+    }
+
```
Comment (BastienBenard) [inline formulaire/src/main/java/fr/openent/formulaire/service/impl/DefaultFolderService.java:55]: Pour guidé les gens a utilisé la version des futures

```suggestion
    /**
     * @deprecated Use {@link #get(String)}
     */
    @Override
    @Deprecated
    public void get(String folderId, Handler<Either<String, JsonObject>> handler) {
           this.get(folderId)
               .onSuccess(res -> handler.handle(new Either.Right(res)))
               .onFailure(error -> handler.handle(new Either.Left(error.getMessage())))
    }

    @Override
    public Future<JsonObject> get(String folderId, UserInfos user) {
        Promise<JsonObject> promise = Promise.promise();
String errorMessage = "[Formulaire@DefaultFolderService::get] Fail to get folder with id " + folderId + " : ";
        String query = "SELECT * FROM " + FOLDER_TABLE + " WHERE id = ?;";
        JsonArray params = new JsonArray().add(folderId);
        Sql.getInstance().prepared(query, params, SqlResult.validUniqueResultHandler(FutureHelper.handlerEither(promise, errorMessage)));

        return promise.future();
    }

```

De plus un model de la table FOLDER_TABLE serait pas mal.
Tu as réellement besoin de l'user en paramètre de fonction ?

File: common/src/main/java/fr/openent/form/helpers/IModelHelper.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
+package fr.openent.form.helpers;
```
Comment (qperie) [inline common/src/main/java/fr/openent/form/helpers/IModelHelper.java:1]: Copié-collé de CRRE

File: common/src/main/java/fr/openent/form/helpers/StringHelper.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
+package fr.openent.form.helpers;
```
Comment (qperie) [inline common/src/main/java/fr/openent/form/helpers/StringHelper.java:1]: Copié-collé de CRRE

**Review by JinDouang — APPROVED** [review]

File: common/src/main/java/fr/openent/form/helpers/IModelHelper.java (hunk 2 above)
Comment (BastienBenard) [reply common/src/main/java/fr/openent/form/helpers/IModelHelper.java:1]: c'est pas le bon
https://github.com/OPEN-ENT-NG/crre/blob/dev/src/main/java/fr/openent/crre/helpers/IModelHelper.java

**Pull request merged**